### DISCUSSION TITLE
Why call method is not returning response?

### DISCUSSION THREAD

**Discussion by hidden-codes** [miguelgrinberg/python-socketio#1368] [Q&A]
I want to receive response for each emit and then send next emit request on the basis of response received for previous emit. I have following code but its not returning anything in response
```
for i in range(1,33):
      payload = "test"+str(i)+"@test.com"
      data = {
                "token" : session_token,
                "email" : payload
      }
      response = sio.call(event_name,data)
      if response == True:
            print("processed")
            break
      else:
            pass
```
I want to check if I received true in emit response from server then do this specific action or if I received false in emit response from server then do this specific action. In short I want to send emit request on the basis of previous emit request or is it possible that all the emit requests are sent once but I would be able to track the response for each emit request?

**Comment by miguelgrinberg** [comment]
How does the other side send the response?

**Comment by hidden-codes** [comment]
server side is based on nodejs. following is the sample code

```
	socket.on('verifyEmailAddress', async function(data) {
		userController.searchByEmailAddress(data.email)
		.then((found) => {
		  socket.emit('isFound', found);
		})
		.catch((error) => {
		  socket.emit('message', {type: "error", message: error.message});
		});
	});
```

**Reply by miguelgrinberg** [reply]
And you see that you are not sending any true/false response here, right? That's what is missing. See the JS docs on how to send [acknowledgements](https://socket.io/docs/v3/emitting-events/#acknowledgements) if you are not familiar with how these responses work.

**Reply by hidden-codes** [reply]
No, either True or False is being returned by userController. Please refer below: (sorry for not attaching earlier)
```
const searchByEmailAddress = async (email) =>  {
	try {
		const users = await models.sequelize.query("SELECT * FROM `user_table` WHERE email LIKE '" + email + "%'", { type: QueryTypes.SELECT });
		if (user.length > 0){
			return true
		}else{
			return false
		}
	  } catch (err) {
		throw new Error("Something went wrong during the query")
	  }
}
```

**Reply by miguelgrinberg** [reply]
One more time, please read about Acknowledgements in the Socket.IO client documentation for JavaScript. The returns that you do in your client-side functions are for your own client-side application, Python has no way to see them.
